modules/seqexec/server/src/main/scala/seqexec/server/package.scala

```diff
@@ -16,21 +16,27 @@ import monocle.Lens
 import monocle.macros.GenLens
 import monocle.function.At.at
 import monocle.function.At.atMap
-import seqexec.engine.{ActionMetadata, ActionMetadataGenerator, Engine}
-import seqexec.model.enum.{ CloudCover, Instrument, ImageQuality, SkyBackground, WaterVapor}
-import seqexec.model.{ Conditions, Observer, Operator, SequenceState }
+import seqexec.engine.Engine
+import seqexec.model.{Conditions, Observer, Operator, SequenceState}
+import seqexec.model.enum.{CloudCover, Instrument, ImageQuality, SkyBackground, WaterVapor}
 import seqexec.model.UserDetails
 
 package server {
   @Lenses
-  final case class EngineMetadata(queues: ExecutionQueues, selected: Map[Instrument, Observation.Id], conditions: Conditions, operator: Option[Operator])
+  final case class ObserverSequence(observer: Option[Observer], seq: SequenceGen)
   @SuppressWarnings(Array("org.wartremover.warts.PublicInference"))
-  object EngineMetadata {
-    implicit val eq: Eq[EngineMetadata] = Eq.by(x => (x.queues, x.selected, x.conditions, x.operator))
+  object ObserverSequence {
+    implicit val eq: Eq[ObserverSequence] = Eq.by(x => (x.observer, x.seq))
+  }
+  @Lenses
+  final case class EngineState(queues: ExecutionQueues, selected: Map[Instrument, Observation.Id], conditions: Conditions, operator: Option[Operator], sequences: Map[Observation.Id, ObserverSequence], executionState: Engine.State)
```

Comment:
There is a one to one relationship between the elements of `sequences` and `executionState.sequences` that I didn't know how to represent without complicating the structure of executionState.